src/test/kotlin/fr/social/gouv/agora/infrastructure/qag/repository/QagInfoRepositoryImplTest.kt

```diff
@@ -257,4 +255,87 @@ internal class QagInfoRepositoryImplTest {
             then(mapper).should(only()).updateQagStatus(dto = qagDTO, qagStatus = QagStatus.MODERATED_ACCEPTED)
         }
     }
+
+    @Nested
+    inner class ArchiveQagTestCases {
+        @Test
+        fun `archiveQag - when invalid UUID - should return FAILURE`() {
+            // When
+            val result = repository.archiveQag(qagId = "invalid UUID")
+
+            // Then
+            assertThat(result).isEqualTo(QagArchiveResult.FAILURE)
+            then(cacheRepository).shouldHaveNoInteractions()
+            then(databaseRepository).shouldHaveNoInteractions()
+            then(mapper).shouldHaveNoInteractions()
+        }
+
+        @Test
+        fun `archiveQag - when cache is not initialized and has no result - should return FAILURE`() {
+            // Given
+            given(cacheRepository.getAllQagList()).willReturn(CacheResult.CacheNotInitialized)
+            given(databaseRepository.getAllQagList()).willReturn(emptyList())
+
+            // When
+            val result = repository.archiveQag(qagId = UUID.randomUUID().toString())
+
+            // Then
+            assertThat(result).isEqualTo(QagArchiveResult.FAILURE)
+            inOrder(cacheRepository, databaseRepository).also {
+                then(cacheRepository).should(it).getAllQagList()
+                then(databaseRepository).should(it).getAllQagList()
+                then(cacheRepository).should(it).initializeCache(emptyList())
+                it.verifyNoMoreInteractions()
+            }
+            then(mapper).shouldHaveNoInteractions()
+        }
+
+        @Test
+        fun `archiveQag - when cache returns DTO - should return SUCCESS`() {
+            // Given
+            val qagUUID = UUID.randomUUID()
+            val qagDTO = mock(QagDTO::class.java).also {
+                given(it.id).willReturn(qagUUID)
+            }
+            given(cacheRepository.getAllQagList()).willReturn(CacheResult.CachedQagList(listOf(qagDTO)))
+            val archivedQagDTO = mock(QagDTO::class.java)
+            given(mapper.archiveQag(dto = qagDTO)).willReturn(archivedQagDTO)
+
+            // When
+            val result = repository.archiveQag(qagId = qagUUID.toString())
+
+            // Then
+            assertThat(result).isEqualTo(QagArchiveResult.SUCCESS)
+            then(databaseRepository).should(only()).save(archivedQagDTO)
+            then(cacheRepository).should().getAllQagList()
+            then(cacheRepository).shouldHaveNoMoreInteractions()
```

Comment:
S'il y a qu'un should, tu peux mettre un `only()`

Reply:
done